Added dialogue

Justin1L8 opened the pull request:
Implemented dialogue sequences / cutscenes, created with `DialogueSequence` scriptable objects and played with `DialoguePlayer` game objects. Dialogue can have multiple characters (each with their own name and image) and can call custom events in the middle of the dialogue (e.g. for characters moving around like a cutscene). For reference of how to write a dialogue sequence, see the example in Assets/Scripts/Dialogue/DialogueL1Start

Justin1L8 committed:
Dialogue first commit

Added dialogue system basics

0GhostLing0 committed:
Dialogue start function (wip)

Justin1L8 committed:
Finished Dialogue System

Fully implemented starting and ending dialogue cutscenes.
(also moved some stuff around. animations are in the Animation folder, temp art is in Art/ProgrammerArt)

Justin1L8 committed:
Tweaked config

Fixed character names in the example cutscene.

Justin1L8 committed:
Merge branch 'main' into Justin-Dialogue

Justin1L8 committed:
Added comments

Added some comments and renamed DialogueEvent to DialogueCommand.

MistOfJudgement reviewed (changes requested):
Pressing space while a hide dialogue couroutine is running causes the unintended behavior where multiple lines are playing at the same time. Probably need a check for when the dialogueplayer is waiting on a couroutine to finish.

Also, not a blocker but unity text boxes support rich text tags like \<b\> or \<color\>. The text writer coroutine displays some of the tag's characters briefly before the tags are applied. I have no clue if writing team would use rich text tags but its there.

Justin1L8 committed:
Fixed bug (overlapping text when pressing space during animation)

During a cutscene where the dialogue box was hidden, you could press space multiple times to play multiple lines of dialogue at a time. Now, pressing space while the box is hidden does nothing.

Justin1L8 commented:
Thanks for the feedback. Fixed the first bug. The second I'll have to look into later if it becomes an issue.
